lib/api/dsl/methods.js

```diff
@@ -856,11 +858,12 @@ function buildCurriedFunction(index, collection, field, operatorName, value, cur
   }
 
   if (!methods.dsl.filtersTree[index][collection].fields[field][curriedFunctionName]) {
-    curriedFunction = _.curry(operators[operatorName]);
-    curriedFunction = _.curry(curriedFunction(field, value));
-    if (not) {
-      curriedFunction = _.negate(curriedFunction);
-    }
+    // we need the ugly eval code to make sure the function can later on be serialized and
```

Comment:
I cannot agree with you on this one. eval is evil, especially using a direct call like we are doing here. But you're right, I cannot figure out a better alternative for the time being.